### ISSUE TITLE
Cloud client builder does not allow changing stages for testing

### ISSUE THREAD

**Issue by npepinpe** [camunda/camunda#7653]
**Describe the bug**

The cloud client builder assumes that the client wants to connect to the production stage. This is fine in most cases, but blocks us from using it for testing. The base address is hard coded to the production stage.

**To Reproduce**

Create a cluster not in production stage, then fill in the following below (replacing the parameters correctly):

```java
final var client = ZeebeClient.newCloudClientBuilder()
        .withClusterId("clusterId")
        .withClientId("clientId")
        .withClientSecret("clientSecret")
        .withRegion("region")
        .build();
final var topology = client.newTopologyCommand().send().join();
```

The `join` will fail since it will try to contact your cluster at `zeebe.camunda.io`, which is the production stage.

**Expected behavior**

I can specify the base address to use so we can test it in different stages.

**Comment by aivinog1** [comment]
@npepinpe Hey! Do you need also the ability to change the auth URL? For now, as I can see is impossible to change just `authorizationServerUrl` only to rewrite `credentialsProvider` completely.

**Comment by npepinpe** [comment]
True, you would also need to change the authorization URL as well. So you'd need to be able to change both. This is already do-able via the OAuthCredentialsProviderBuilder, but you're right that you'd now need to rebuild it from scratch, which defeats the purpose of the cloud client builder.

**Comment by korthout** [comment]
Marking this as a feature because it enhances the cloud client builder's capabilities.

**Comment by korthout** [comment]
Marking priority as `later later` because there is a workaround available.

Please comment if you think this should have a higher priority.
